docs/cookbooks/creating_a_derived_dataset.md

```diff
@@ -0,0 +1,208 @@
+  metrics.string.usage_variant AS mozregression_variant,
+  metrics.string.usage_app AS app_used,
+  normalized_os AS os,
+  mozfun.norm.truncate_version(normalized_os_version, "minor") AS os_version,
+  count(DISTINCT(client_info.client_id)) AS distinct_clients,
+  count(*) AS total_uses
+FROM
+  `moz-fx-data-shared-prod`.org_mozilla_mozregression.usage
+WHERE
+  DATE(submission_timestamp) = @submission_date
+  AND client_info.app_display_version NOT LIKE '%.dev%'
+GROUP BY
+  date,
+  mozregression_version,
+  mozregression_variant,
+  app_used,
+  os,
+  os_version;
+```
+
+We use the `truncate_version` UDF to omit the patch level for MacOS and Linux, which should both reduce the size of the dataset as well as make it more difficult to identify individual clients in an aggregated dataset.
+
+We also have a short clause (`client_info.app_display_version NOT LIKE '%.dev%'`) to omit developer versions from the aggregates: this makes sure we're not including people developing or testing mozregression itself in our results.
+
+## Formatting and validating the query
+
+Now that we've written our query, we can format it and validate it. Once that's done, we run:
+
+```bash
+./bqetl query validate org_mozilla_mozregression_derived.mozregression_aggregates
+```
+
+If there are no problems, you should see no output.
+
+## Creating a DAG
+
+BigQuery-ETL has some facilities in it to automatically add your query to [telemetry-airflow](https://github.com/mozilla/telemetry-airflow) (our instance of Airflow).
+
+Before scheduling your query, you'll need to find an [Airflow DAG](https://airflow.apache.org/docs/apache-airflow/stable/concepts.html#dags) to run it off of. In some cases, one may already exist but in this one, we don't have one -- so we'll create it:
+
+```bash
+./bqetl dag create bqetl_mozregression --schedule-interval "0 4 * * *" --owner wlachance@mozilla.com --description "This DAG schedules queries for populating queries related to Mozilla's internal developer tooling (e.g. mozregression)." --start-date 2020-06-01
+```
+
+Most of the options above should be self-explanatory. We use a schedule interval of "0 4 \* \* \*" (4am UTC daily) instead of "daily" (12am UTC daily) to make sure this isn't competing for slots with desktop and mobile product ETL.
+
+## Scheduling your query
+
+Once again, you access this functionality via the `bqetl` tool:
+
+```bash
+./bqetl query schedule org_mozilla_mozregression_derived.mozregression_aggregates_v1 --dag bqetl_mozregression --task-name mozregression_aggregates__v1
+```
+
+Note that we are scheduling the generation of the underlying _table_ which is `org_mozilla_mozregression_derived.mozregression_aggregates_v1` rather than the view.
+
+After doing this, you will also want to generate the actual airflow configuration which telemetry-airflow will pick up. Run:
+
+```bash
+./bqetl dag generate
+```
+
+This may take a while, as it currently does a dry run through every query defined in bigquery-etl.
```

Comment:
`bqetl dag generate bqetl_mozregression` would be faster here ;)

Reply:
Does that work? I'll adjust it if so. 😄

Reply:
It does!